envoy/http/query_params.h

```diff
@@ -15,6 +18,22 @@ namespace Utility {
 using QueryParams = std::map<std::string, std::string>;
 using QueryParamsVector = std::vector<std::pair<std::string, std::string>>;
 
+class QueryParamsMulti {
+private:
+  std::map<std::string, std::vector<std::string>> data;
+
+public:
+  void remove(std::string key);
```

Comment:
absl::string_view rather than std::string for `remove`, `overwrite` and `add`? to avoid the copy during the argument passing

Reply:
Please bear with me here, because my C++ is rusty at best.

I swapped the `key` parameters to `absl::string_view`s without an issue.  This makes perfect sense - thank you!

I don't think I can safely store `string_view`s in the map, since I have no way to guarantee that the data underlying the `string_view` won't change.

I couldn't come up with a way to avoid copying the `string_view` into a `string` when calling either `add` or `overwrite`, or forming a vector with the string in it.  Typing the parameter as `std::string` seemed performance-equivalent and far easier to read to me.

I tried a bunch of things along the lines of:
```c++
void add(absl::string_view key, absl::string_view value) {
  this->data[key] = std::vector<std::string>{value}; 
}
```
except everything I tried to get `value` to a `std::string` to initialize the vector with made the compiler angry. 

If I'm incorrect, and there is a performance optimization here I'm missing, any pointers you could provide about what the code should look like would be great!

Reply:
Yes, pass by string_view (or const &) and extend its lifetime if needed like storing into container is what I meant.
Thanks!